### PR TITLE
[FEATURE] Fixes #205. Added Host management feature.

### LINKED ISSUE #205: Add support for host management and monitoring.
To support the new APIs for Host Management and monitoring. The basic APIs that will be supported are:  

- Create / Register a host  
- List all hosts  
- Get a single host details  
- Modify host info/details.  

Reference Link: [Click here](https://docs.google.com/document/d/1VoAHlZwQIubNCXCGeFdZjnXLSq7iqChHF_Ir6Y07o1E/edit) for the design doc.

### PR DESCRIPTION
**Pull request by anvithks**
Added the host management feature.  
User can perform the following:  
- Add an existing Agentless host  
- View the list of hosts available.  
- View the details of a single host  
- Expand the details in the host list.  
- Delete a single host.  
- Batch delete multiple hosts.  
- List Attached volumes and Initiators
- Modify host (To be fixed in #213 )

Screenshots:  
List Hosts  
list-hosts

Add Host  
create-host

Add Host Basic Validations  
create-host-validation

Expand Host Details  
exapnd-details

Host Details  
host-deatils

### CODE REVIEW

**Comment by anvithks** [comment]
@wisererik and @kumarashit Please review the PR.
Modify host feature will be added in #213

**Comment by anvithks** [comment]
In list host page the overflowing availability zones issue seen in the screenshots above  is fixed.

File: src/app/business/block/create-host/create-host.component.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        required: "IP Address is required",
+        pattern: "Enter valid IP address"
+    },
+    "hostName": {
+        required: "Host Name is required.",
+        maxlength: "Maximum 28 characters",
+        minlength: "Minimum 2 characters"
+    }
+  };
+  validRule = {
```
Comment (kumarashit) [inline src/app/business/block/create-host/create-host.component.ts]: Put a comment it's for IPV4

File: src/app/business/block/hosts.component.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+                    }
+                },
+                disabled:true
+            }
+        ];
+
+    }
+
+    getAllHosts(){
+       /*  this.HostsService.getHosts().subscribe((res) => {
```
Comment (kumarashit) [inline src/app/business/block/hosts.component.ts]: remove if not required

File: src/app/business/block/hosts.component.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+                })
+
+            },
+            reject:()=>{}
+        })
+
+    }
+
+
+  /*   modifyVolume() {
```
Comment (kumarashit) [inline src/app/business/block/hosts.component.ts]: This is not relevant, right? If yes, please remove this

**Comment by kumarashit** [comment]
One generic comment for the UI, we are using 'Please' word everywhere like "Please select host" "Please choose a protocol"..IMO, we should avoid using 'please' word unless we are asking some inconvenient input from user.
Just a thought

File: src/app/i18n/en/keyID.json [hunk 4]
```diff
@@ -142,5 +143,15 @@
     "sds_lifeCycle_disable": "Disable the lifecycle rule",
     "sds_fileShare_export": "Export Location",
     "sds_fileShare_update": "Updated At",
-    "sds_ippattern": "IP address must be (1~255). (0~255). (0~255). (0~255)/(1-32)"
+    "sds_ippattern": "IP address must be (1~255). (0~255). (0~255). (0~255)/(1-32)",
```
[the remark this answers is not in the document]
Comment (anvithks) [reply src/app/i18n/en/keyID.json]: This is a older keyID and not used in this current page. FOr the current page I will add the message.

File: src/app/business/block/create-host/create-host.component.ts (hunk 1 above)
Comment (anvithks) [reply src/app/business/block/create-host/create-host.component.ts]: ok.

File: src/app/business/block/hosts.component.ts (hunk 2 above)
Comment (anvithks) [reply src/app/business/block/hosts.component.ts]: ok.

File: src/app/business/block/hosts.component.ts (hunk 3 above)
Comment (anvithks) [reply src/app/business/block/hosts.component.ts]: ok.

**Comment by anvithks** [comment]
> One generic comment for the UI, we are using 'Please' word everywhere like "Please select host" "Please choose a protocol"..IMO, we should avoid using 'please' word unless we are asking some inconvenient input from user.
> Just a thought

Generally it is to keep the tone polite. It is again a perception thing whether we consider saying please only for an inconvenience. We can review the overall messages and tone when we start the rewrite of the dashboard. I will add this is a ToDo for the design phase.

**Comment by skdwriting** [comment]
@anvithks how did we test these dashboard changes? Can we keep a clear test information on this?

**Comment by skdwriting** [comment]
> @anvithks how did we test these dashboard changes? Can we keep a clear test information on this?

We do not have the backend ready yet right?

**Comment by anvithks** [comment]
> @anvithks how did we test these dashboard changes? Can we keep a clear test information on this?

@skdwriting These are tested with @wisererik s local dev environment. They have not been tested with a backend created from development branch yet.

**Comment by anvithks** [comment]
@kumarashit @skdwriting Please review and merge

**Comment by kumarashit** [comment]
LGTM

File: src/app/business/block/hosts.component.html [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                        <p-column field="portName" header="Port Name"></p-column>
+                        <p-column field="protocol" header="Protocol"></p-column>
+                    </p-dataTable>
+                </div>
+            </div>
+    </ng-template>
+</p-dataTable>
+<!--create volumes group-->
+<!-- <p-dialog header="{{I18N.keyID['sds_block_volume_group_create_group']}}" [(visible)]="showHostsDialog" [modal]="true" [responsive]="true" [width]="800" [contentStyle]="{'min-height':'300px'}">
+    <form [grid]="{label: 'ui-g-2', content:'ui-g-10'}" [formGroup]="volumeGroupForm" [errorMessage]="errorMessage">
```
Comment (wisererik) [inline src/app/business/block/hosts.component.html]: please remove unused code here

**Pull request merged**